app/api-v1/controllers/Attachment/index.js

```diff
@@ -1,5 +1,72 @@
+const { parseAccept } = require('../../../utils/parseAcceptHeader')
+const logger = require('../../../utils/Logger')
+
+const db = require('../../../db')
+
+const { createAttachmentFromFile, createAttachment } = require('../Attachment/helpers')
+const { InternalError, NotFoundError, BadRequestError } = require('../../../utils/errors')
+
 module.exports = {
-  getAll: async function () {
-    return { status: 500, response: { message: 'Not Implemented' } }
+  get: async (req) => {
+    const { id } = req.params
+    if (!id) throw new BadRequestError('missing params')
+
+    const [attachment] = await db.getAttachment(req.params.id)
+    if (!attachment) throw new NotFoundError('Attachment Not Found')
+    const orderedAccept = parseAccept(req.headers.accept)
+
+    for (const mimeType of orderedAccept) {
+      if (mimeType === 'application/octet-stream' && attachment.filename.includes('json'))
+        return {
+          status: 406,
+          response: 'some error msg',
+        }
+      if (mimeType === 'application/json' || mimeType === 'application/*' || mimeType === '*/*') {
+        const json = JSON.parse(attachment.binary_blob)
+        //It looks like the validator is still the issue, I get a status f 200 befre validatioo
+        return {
+          status: 200,
+          response: json,
+        }
+      }
+      if (mimeType === 'application/octet-stream') {
+        return {
+          status: 201,
```

Comment:
200?

Reply:
yes this is a get and should be a 200

Reply:
this was the tricky part with the content, it was not working with openApi validator as it was taking content whatever is first in order. So when we changed order of from octa to application/json then it was always app/json

Not sure if this is due to mock express or the way we do it. but the main reason was to not get stuck on this and gather some info e.g. looking at their docs:)

Reply:
Is this just a shortcoming of the OpenAPI validator we're using?